src/algebra/polynomial/basic.lean

```diff
@@ -0,0 +1,60 @@
+/-
+Copyright (c) 2020 Aaron Anderson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Aaron Anderson, Jalex Stark.
+-/
+import data.polynomial
+open polynomial finset
+
+/-!
+# Polynomials
+
+Eventually, much of data/polynomial.lean should land here.
+
+## Main results
+
+We define two `ring_hom`s of type `polynomial R →+* R`,
+- `eval_ring_hom`
+- `coeff_zero_ring_hom`
+
+We define a `monoid_hom` of type `polynomial R →* R`,
+- `leading_coeff_monoid_hom`
+-/
+
+universes u w
+
+variables {R : Type u} {α : Type w}
+
+namespace polynomial
+
+section comm_semiring
+variables [comm_semiring R]
+
+/-- A `ring_hom` which evaluates polynomials at a particular value -/
+noncomputable def eval_ring_hom : R → (polynomial R →+* R) := eval₂_ring_hom (ring_hom.id R)
```

Comment:
Please use `noncomputable theory` at the top of the file, instead of individual `noncomputable` modifiers.